### PR TITLE
Fix A bug for Metric Generator Test

### PR DESCRIPTION
**Pull request by XinRanZhAWS**
Previously, testBothMetricsWhenLocalRootConsumerProcess will always assert true
Description of changes:
test generated attributes if it's not null

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by thpierce — APPROVED** [review]

File: .github/workflows/appsignals-e2e-ec2-test.yml [hunk 1]
```diff
@@ -43,7 +44,7 @@ jobs:
       - name: Set CW Agent RPM environment variable
         run: |
           if [ ${{ env.AWS_DEFAULT_REGION }} == "us-east-1" ]; then
-            echo GET_CW_AGENT_RPM_COMMAND="wget -O cw-agent.rpm https://amazoncloudwatch-agent-us-east-1.s3.amazonaws.com/amazon_linux/amd64/1.300031.0b313/amazon-cloudwatch-agent.rpm" >> $GITHUB_ENV
+            echo GET_CW_AGENT_RPM_COMMAND="wget -O cw-agent.rpm https://github.com/XinRanZhAWS/amazon-cloudwatch-agent/releases/download/test/amazon-cloudwatch-agent.rpm" >> $GITHUB_ENV
```
Comment (vasireddy99) [inline .github/workflows/appsignals-e2e-ec2-test.yml:47]: Is this expected to have it referred to the personal fork.

File: .github/workflows/pr-build.yml [hunk 2]
```diff
@@ -4,6 +4,7 @@ on:
     branches:
       - main
       - "release/v*"
+  workflow_dispatch:
```
Comment (vasireddy99) [inline .github/workflows/pr-build.yml:7]: Do we want to run PR build in workflow_dispatch, as run async other than at the time of PR.

File: .github/workflows/pr-build.yml [hunk 3]
```diff
@@ -81,6 +82,10 @@ jobs:
         with:
           arguments: testing:validator:build
 
+      - name: Dry Run Validator
```
Comment (vasireddy99) [inline .github/workflows/pr-build.yml:85]: Is the idea to catch compilation failures as well, also can you please check if it needs to run all os [reference here](https://github.com/aws-observability/aws-otel-java-instrumentation/pull/738/files#diff-404033996c34f9971e0c112d8538b9063618fc85059a44df7d4db448974888caR81).

File: .github/workflows/appsignals-e2e-ec2-canary-test.yml [hunk 4]
```diff
@@ -4,8 +4,8 @@
 ## including logs, metrics, and traces.
 name: App Signals Enablement - E2E EC2 Canary Testing
 on:
-  schedule:
-    - cron: '0/15 * * * *' # run the workflow every 15 minutes
+#   schedule:
```
Comment (vasireddy99) [inline .github/workflows/appsignals-e2e-ec2-canary-test.yml:7]: Will commenting this trigger alarms.

**Comment by thpierce** [comment]
PR very out of date.

**Pull request closed without merging**